src/MessageCreationUtil.js

```diff
@@ -32,16 +31,14 @@ export default class MessageCreationUtil {
 
     async getStream(streamId) {
         if (!this.cachedStreams.get(streamId)) {
-            if (!this.ongoingRequests[streamId]) {
-                this.ongoingRequests[streamId] = this.getStreamFunction(streamId)
-            }
-            const stream = await this.ongoingRequests[streamId]
-            const success = this.cachedStreams.set(streamId, {
+            const streamPromise = this.getStreamFunction(streamId).then((stream) => ({
                 id: stream.id,
                 partitions: stream.partitions,
-            })
+            }))
```

Comment:
Nice one, immediately cache the promise not the stream and avoid any timing issues. Nice.

Reply:
Does `node-cache`'s `useClones: true` affect this in any way?

Reply:
I guess it wouldn't hurt to set it to `false` since we don't modify the values after retrieving them from the cache. But probably safer to keep it to `true` unless we need cache a complex mutable object, the performance cost of cloning should be very small for this object (only 2 fields)

Reply:
the promise is what's stored in the cache so it's "cloning" the promise (whatever that means), not the object with two fields, right?

Reply:
Yes the promise actually

Reply:
Can cloning promises even work? When the original promise is resolved, is the clone resolved too? I feel slightly uneasy just thinking of a cache returning cloned promises on `get`.